Added Sort Command

bladerail opened the pull request:
Added a Sort Command that currently sorts the list by name permanently. The original sorting order of date added is lost. Will aim to implement further sort types.

bladerail committed:
Merge branch 'master' of https://github.com/CS2103AUG2017-F11-B2/main into W8Sort

bladerail committed:
Travis notifications removed

bladerail committed:
Merge branch 'master' of https://github.com/CS2103AUG2017-F11-B2/main

bladerail committed:
Merge remote-tracking branch 'origin/master' into W8Sort

bladerail committed:
Taught the app to accept Sort command but does nothing
. Changed alias of select to 's' to allow Sort to have the alias 's'

bladerail committed:
SortCommand now accepts arguments, but still does nothing.

. Modified SortCommand to take in a String filterType as input, then print the inputs as an error message
. Modified SortCommandTest to test the equals method
. Added SortCommandParser that parses input and checks if there is one or zero arguments
. Modified AddressBookParser to use the SortCommandParser
. Modified AddressBookParserTest to ensure that user input generates the correct command
. Added SortCommandParserTest to test boundary values for SortCommandParser

bladerail committed:
Travis

bladerail committed:
Travis fixes

bladerail committed:
Added SortCommand Logic that is throwing exceptions unintentionally
. Added sortFilteredPersonList method in ModelManager class
. Added sortFilteredPersonList method in Model interface

bladerail committed:
Implemented SortCommand Logic which now correctly sorts the list
. Modified AddressBook.java to include a sortPersons method, which sorts it's persons list
. Modified ModelManager.java to call sortPersons on it's addressBook
. Modified UniquePersonList.java to implement the sortPersons method
. Modified AddCommandTest to include the sortFilteredPersons method in it's stubModel

bladerail committed:
Added Junit Tests for the Sort Command.
. Modified TypicalPersons.java to have 2 different Typical AddressBooks, one with a sorted list and the other unsorted. To prevent confusion, I had to rename every method call of getTypicalAddressBook() to getSortedTypicalAddressBook().
. Modified UniquePersonListTest.java to test for correct sorting.
. Modified ModelManagerTest.java to test for sorting.
. Modified AddressBookTest.java to test for sorting.
. Modified SortCommandTest.java to test for correct command execution.
. Added a comment in SortCommandParserTest for further development

bladerail committed:
Merge branch 'master' of https://github.com/CS2103AUG2017-F11-B2/main into W8Sort

bladerail committed:
Updated Documentation
. Updated UserGuide.adoc to reflect the sort function
. Updated joshua.adoc to reflect changes made
. Made minor changes to several classes which were failing the travis builds

bladerail committed:
Made minor changes to resolve Travis Build Error

hansiang93 commented on src/main/java/seedu/address/logic/commands/SortCommand.java:
```diff
@@ -0,0 +1,55 @@
+            return true;
+        }
+
+        //instanceOf handles nulls
```
Spacing of comments inconsistent

'// comment' is standard over '//comment'

hansiang93 commented on src/main/java/seedu/address/logic/commands/SortCommand.java:
```diff
@@ -0,0 +1,55 @@
+            return false;
+        }
+
+        //state check
```
Same as previous comment

hansiang93 commented on src/main/java/seedu/address/logic/parser/SortCommandParser.java:
```diff
@@ -0,0 +1,39 @@
+        filterType = argMultimap.getPreamble();
+
+        if (filterType.equals("sort")) {
+            //No arguments, set to default
```
same here

hansiang93 commented on src/main/java/seedu/address/logic/parser/SortCommandParser.java:
```diff
@@ -0,0 +1,39 @@
+        }
+
+        if (!(filterType.equals("name") || filterType.equals("default"))) {
+            //Use ParserUtil to filter later?
```
same here

bladerail committed:
Minor edits to files to follow coding standards

bladerail committed:
Merge branch 'master' of https://github.com/CS2103AUG2017-F11-B2/main into W8Sort

bladerail committed:
Modified FilterCommandTest
. Changed getTypicalAddressBook to getSortedTypicalAddressBook, a change required as a result of the Sort Command
